daemon/create_windows.go

```diff
@@ -28,66 +28,76 @@ func (daemon *Daemon) createContainerOSSpecificSettings(container *container.Con
+		v, err := daemon.volumes.Create(
+			context.TODO(),
+			stringid.GenerateNonCryptoID(),
+			hostConfig.VolumeDriver,
+			volumeopts.WithCreateReference(container.ID))
 		if err != nil {
 			return err
 		}
 
-		// FIXME Windows: This code block is present in the Linux version and
-		// allows the contents to be copied to the container FS prior to it
-		// being started. However, the function utilizes the FollowSymLinkInScope
-		// path which does not cope with Windows volume-style file paths. There
-		// is a separate effort to resolve this (@swernli), so this processing
-		// is deferred for now. A case where this would be useful is when
-		// a dockerfile includes a VOLUME statement, but something is created
-		// in that directory during the dockerfile processing. What this means
-		// on Windows for TP5 is that in that scenario, the contents will not
-		// copied, but that's (somewhat) OK as HCS will bomb out soon after
-		// at it doesn't support mapped directories which have contents in the
-		// destination path anyway.
-		//
-		// Example for repro later:
-		//   FROM windowsservercore
-		//   RUN mkdir c:\myvol
-		//   RUN copy c:\windows\system32\ntdll.dll c:\myvol
-		//   VOLUME "c:\myvol"
-		//
-		// Then
-		//   docker build -t vol .
-		//   docker run -it --rm vol cmd  <-- This is where HCS will error out.
-		//
-		//	// never attempt to copy existing content in a container FS to a shared volume
-		//	if v.DriverName() == volume.DefaultDriverName {
-		//		if err := container.CopyImagePathContent(v, mp.Destination); err != nil {
-		//			return err
-		//		}
-		//	}
-
-		// Add it to container.MountPoints
-		container.AddMountPointWithVolume(mp.Destination, &volumeWrapper{v: v, s: daemon.volumes}, mp.RW)
+		// Add it to container.MountPoints. Note that the last parameter is true - as in read-write
+		container.AddMountPointWithVolume(destination, &volumeWrapper{v: v, s: daemon.volumes}, true)
 	}
+
+	// NOTE: On Linux, this function returns daemon.populateVolumes(container).
+	// This allows the contents of a volume to be copied to the containers
+	// filesystem prior to it be in started.
+	//
+	// There are many issues with solving this problem, and the reality is
+	// that it will likely be one thing that can't be done on Windows.
+	//
+	// FollowSymLinkInScope doesn't cope with Windows volume-style file paths.
+	// To avoid break outs we need to do scoped access. This is surmountable.
+	//
+	// Argons might be possible, but at this execution point on Windows,
+	// the container filesystem isn't mounted as it's done much later than
+	// on Linux.
+	//
+	// Xenons (both WCOW and LCOW) are difficult as we don't want to mount
+	// the filesystem on the host, for both security and perf reasons.
+	// Further, obviously the LCOW filesystem can't be mounted on the host
+	// directly.
+	//
+	// What this means is that on Windows, the contents of a VOLUME created
+	// in a Dockerfile with contents will NOT be copied.
```

Comment:
Is this limitation only the case when using a `VOLUME` in a `Dockerfile`, or also for `docker run`? (i.e. `docker run -v somevolume:/some/path`). If the same applies to `docker run`, that's something we should clearly document (if it's not documented yet), as that might be a big issue for quite some use-cases.


For the Dockerfile case, I don't think it _has to_ be a big issue (the behavior in that case is not super-clearly defined)